HandyHansel/BotDatabase/Models/GuildBackgroundJob.cs

```diff
@@ -23,9 +23,16 @@ public class GuildBackgroundJob
         [Column("job_type")]
         public GuildJobType GuildJobType { get; set; }
 
-        public void ConvertTimeZoneTo(TimeZoneInfo timezone)
+        public GuildBackgroundJob WithTimeZoneConvertedTo(TimeZoneInfo timezone)
         {
-            this.ScheduledTime = TimeZoneInfo.ConvertTimeFromUtc(this.ScheduledTime, timezone);
+            return new GuildBackgroundJob
+            {
+                HangfireJobId = this.HangfireJobId,
+                GuildId = this.GuildId,
+                JobName = this.JobName,
+                ScheduledTime = TimeZoneInfo.ConvertTimeFromUtc(this.ScheduledTime, timezone),
+                GuildJobType = this.GuildJobType,
+            };
```

Comment:
Switch to a functional paradigm that doesn't rely on side effects (because the side effects break the database.)